snapm/manager/plugins/lvm2.py

```diff
@@ -298,11 +310,36 @@ def is_lvm_device(devpath):
     return uuid.startswith(LVM_UUID_PREFIX)
 
 
+def vg_lv_from_blockdev(devpath):
+    """
+    Return a ``(vg_name, lv_name)`` tuple for the LVM device at
+    ``devpath``.
+    """
+    dmsetup_cmd_args = [
+        LVS_CMD,
+        LVS_NO_HEADINGS,
+        "-o",
+        LVS_FIELD_MIN_OPTIONS,
+        LVS_DEV_SELECTOR,
+        LVS_DEV_PATH + devpath,
+    ]
+    try:
+        dmsetup_cmd = run(dmsetup_cmd_args, capture_output=True, check=True)
+    except CalledProcessError as err:
+        raise SnapmCalloutError(f"Error calling {DMSETUP_CMD}") from err
+    name = dmsetup_cmd.stdout.decode("utf8").strip()
+    name_parts = name.split(" ")
+    return (name_parts[0], name_parts[1])
+
 def vg_lv_from_device_path(devpath):
     """
     Return a ``(vg_name, lv_name)`` tuple for the LVM device at
     ``devpath``.
     """
+
+    if S_ISBLK(stat(devpath).st_mode):
+        return vg_lv_from_blockdev(devpath)
+
```

Comment:
This stat expression should always return `True` for valid LVM device paths - whether they're in `/dev/mapper/VG-LV` or `/dev/VG/LV` style, the stat will follow the symlink to a dm-N block device node and return `vg_lv_from_blockdev()`, but that will only work if the path is in `/dev/VG/LV` form.

The original function is not very well named but its job is to turn `/dev/mapper/...` device paths (that we get from a lookup in `/proc/self/mounts`) into a (VG, LV) tuple.

I think we can simplify this and have a single function to translate either style by using `lvs` and the `LV` argument - this accepts either a `/dev/mapper` path or a `/dev/VG/LV` path and outputs the VG/LV pair:

```
# lvs -ovg_name,lv_name /dev/mapper/fedora-test1
  VG     LV   
  fedora test1
# lvs -ovg_name,lv_name /dev/fedora/test1
  VG     LV   
  fedora test1
```